dace/transformation/dataflow/map_distribution.py

```diff
@@ -0,0 +1,940 @@
+        
+        for desc, accesses in outputs.items():
+            if isinstance(desc, (dace.data.Array, dace.data.View)):
+                for a in accesses:
+                    if a.num_elements() != 1:
+                        return False
+                    indices = a.min_element()
+                    unmatched_indices = set(params)
+                    for idx in indices:
+                        if idx in unmatched_indices:
+                            unmatched_indices.remove(idx)
+                    if len(unmatched_indices) > 0:
+                        return False
+            else:
+                return False
+
+        return outer_product_found
+
+    @staticmethod
+    def match_to_str(graph: dace.SDFGState, candidate: Dict[pm.PatternNode,
+                                                            int]) -> str:
+        map_entry = graph.node(candidate[OuterProductOperation.map_entry])
+        return map_entry.map.label + ': ' + str(map_entry.map.params)
+
+    def apply(self, sdfg: dace.SDFG):
+        pass
+
+
+@registry.autoregister_params(singlestate=True)
+class Reduction1Operation(pm.Transformation):
+    """ Detects reduction1 operations.
+    """
+
+    map_entry = pm.PatternNode(nodes.MapEntry)
+
+    @staticmethod
+    def expressions():
+        return [sdutil.node_path_graph(Reduction1Operation.map_entry)]
+
+    @staticmethod
+    def can_be_applied(graph: dace.SDFGState,
+                       candidate: Dict[pm.PatternNode, int],
+                       expr_index: int,
+                       sdfg: dace.SDFG,
+                       strict: bool = False):
+
+        map_entry = graph.node(candidate[Reduction1Operation.map_entry])
+        map_exit = graph.exit_node(map_entry)
+        params = [dace.symbol(p) for p in map_entry.map.params]
+
+        outputs = dict()
+        for _, _, _, _, m in graph.out_edges(map_exit):
+            if not m.wcr:
+                return False
+            desc = sdfg.arrays[m.data]
+            if desc not in outputs.keys():
+                outputs[desc] = []
+            outputs[desc].append(m.subset)
+        
+        for desc, accesses in outputs.items():
+            if isinstance(desc, dace.data.Scalar):
+                continue
+            elif isinstance(desc, (dace.data.Array, dace.data.View)):
+                for a in accesses:
+                    if a.num_elements() != 1:
+                        return False
+            else:
+                return False
+
+        return True
+
+    @staticmethod
+    def match_to_str(graph: dace.SDFGState, candidate: Dict[pm.PatternNode,
+                                                            int]) -> str:
+        map_entry = graph.node(candidate[Reduction1Operation.map_entry])
+        return map_entry.map.label + ': ' + str(map_entry.map.params)
+
+    def apply(self, sdfg: dace.SDFG):
+        pass
+
+
+@registry.autoregister_params(singlestate=True)
+class ReductionNOperation(pm.Transformation):
```

Comment:
Same as above. I'd rather have those in separate files/PRs if we don't use them now.